### PR TITLE
Add save and clear keychain actions

### PR DESCRIPTION
**Pull request by vinothvino42**
In this PR, you can store the data securely in Keychain(iOS) or Keystore(Android) like below

**ACTION:**
```yaml
saveKeychain:
  key: name
  value: EnsembleUI
  onComplete: |
    //@code
    console.log("Data Saved Successfully");
  onError: |
    //@code
    console.log("Failed to save data");
```

```yaml
clearKeychain:
  key: name
  onComplete: |
    //@code
    console.log("Data Saved Successfully");
  onError: |
    //@code
    console.log("Failed to save data");
```

**CODE:**

```javascript
// To save the value
ensemble.saveKeychain({
  'key': 'name',
  'value': 'EnsembleUI'
});

// To clear the value
ensemble.clearKeychain({'key': 'name'});
```

### CODE REVIEW

File: lib/ensemble_app.dart [hunk 1]
```diff
@@ -25,6 +25,7 @@ import 'package:flutter_localizations/flutter_localizations.dart';
 import 'package:workmanager/workmanager.dart';
 
 const String backgroundUploadTask = 'backgroundUploadTask';
+const String ensembleMethodChannelName = 'com.ensembleui.host.platform';
```
Comment (vusters) [inline lib/ensemble_app.dart:28]: put this somewhere else. This is the EntryPoint to our app and should have just the minimum possible.

Comment (vusters) [reply lib/ensemble_app.dart:28]: should you call it `defaultMethodChannelName`?

File: lib/framework/action.dart [hunk 2]
```diff
@@ -741,7 +741,9 @@ enum ActionType {
   getPhoneContacts,
   checkPermission,
   saveToKeychain,
+  saveToKeychainWithGroupId,
```
Comment (vusters) [inline lib/framework/action.dart]: why? our Actions will balloon to an unreasonable number. GroupId is simply another attribute

**Review by vusters — CHANGES_REQUESTED** [review]

File: lib/ensemble_app.dart (hunk 1 above)
Comment (vinothvino42) [reply lib/ensemble_app.dart:28]: > should you call it `defaultMethodChannelName`?

Okay, I'll change that, can you tell me where to place this constant?

**Review by vusters — APPROVED** [review]

**Pull request merged**